project-koku/koku
# COST-3050: Update deduplicate reports function for OCI

## Jira Ticket

[COST-3050](https://issues.redhat.com/browse/COST-3050)

## Description

This change will update the deduplicate function to merge reports with duplicate manifest ids but different (start, end) date pairs

## Testing

1. Checkout Branch
2. Restart Koku
3. Load some oci data with `make create-test-customer` and `make load-test-customer-data test_source=oci`
4. Hit the sources endpoint to get the oci source id, http://127.0.0.1:8000/api/cost-management/v1/sources/
5. Hit endpoint  `http://127.0.0.1:8000/api/cost-management/v1/sources/<id>/stats/`
    1. You should see unique timestamps in `last_manifest_complete_date` field for each manifest. 

## Notes

...
To test with a real/remote source, you need OCI credentials in your .env file and run 

```
 make oci-source oci_name=<source_name> bucket=<bucket_name> namespace=<namespace> region=<region>

```

and follow steps **4** and **5**
